crates/frontend/src/pages/context_override.rs

```diff
@@ -172,71 +167,88 @@ pub fn context_override() -> impl IntoView {
-        let context_with_mandatory_dimensions = dimensions
-            .into_iter()
-            .filter_map(|dim| {
-                if dim.mandatory {
-                    Some(Condition {
-                        left_operand: dim.dimension,
-                        operator: ConditionOperator::Is,
-                        right_operand: vec![Value::from("")],
-                    })
-                } else {
-                    None
-                }
-            })
-            .collect::<Vec<Condition>>();
-        set_selected_data.set(Some(Data {
-            context: context_with_mandatory_dimensions,
+        let mut default_ctx: Conditions = Conditions(vec![]);
+        for dim in dimensions.iter().filter(|v| v.mandatory) {
+            let r#type = SchemaType::try_from(dim.schema.clone());
+            if let Err(_) = r#type {
+                //TODO emit an alert and return
+                return;
+            }
+
+            let condition = Condition::try_from((
+                Operator::Is,
+                dim.dimension.clone(),
+                r#type.unwrap(),
+            ));
+            if let Err(_) = condition {
+                //TODO emit and alert and return
+                return;
+            }
+
+            default_ctx.push(condition.unwrap());
+        }
+
+        selected_context_ws.set(Some(Data {
+            context: default_ctx,
             overrides: vec![],
         }));
         open_drawer("context_and_override_drawer");
     });
 
-    let handle_submit = Callback::new(move |_| {
+    let on_submit = Callback::new(move |_| {
         close_drawer("context_and_override_drawer");
         set_form_mode.set(None);
-        set_selected_data.set(None);
+        selected_context_ws.set(None);
         page_resource.refetch();
     });
 
-    let handle_context_edit =
-        Callback::new(move |data: (Context, Map<String, Value>)| {
-            let (context, overrides) = data;
-            let conditions = extract_conditions(&Value::Object(context.condition.into()));
-
-            set_selected_data.set(Some(Data {
-                context: conditions,
-                overrides: overrides.into_iter().collect::<Vec<(String, Value)>>(),
-            }));
-            set_form_mode.set(Some(FormMode::Edit));
-
-            open_drawer("context_and_override_drawer");
-        });
+    let on_context_edit = Callback::new(move |data: (Context, Map<String, Value>)| {
+        let (context, overrides) = data;
+        match Conditions::from_context_json(&context.condition.into()) {
+            Ok(conditions) => {
+                selected_context_ws.set(Some(Data {
+                    context: conditions,
+                    overrides: overrides.into_iter().collect::<Vec<(String, Value)>>(),
+                }));
+                set_form_mode.set(Some(FormMode::Edit));
+                open_drawer("context_and_override_drawer");
+            }
+            Err(e) => {
```

Comment:
At this point in the program, if this call errs out, there's no useful error we can report to the user to un-block them. So maybe, it makes sense to use some kind of **front-end panic** which let's us write code as if there was no chance of an error to begin with. What do you think?